### ISSUE TITLE
IGEMM cannot handle model-switch between training/eval

### ISSUE THREAD

**Issue by blackencino** [openvdb/fvdb-core#9]
torch.nn.Module has the ability to be switched between training mode and eval mode. Modules are created in "training" mode by default, but can be switched into eval mode:

```
model.eval()
```
or back:
```
model.train()
```

The status is captured by a variable on the module, `self.training`. 

In `SparseConvPackInfo.cpp`, we have a function which is called to create the backend acceleration structure. When the backend is set to IGEMM, the code is called with the `self.training` flag:

```
elif backend == "igemm_mode0":
            kmap.build_implicit_gemm(
                sorted=False, split_mask_num=1, training=self.training, split_mask_num_bwd=3, use_tf32=self.allow_tf32
            )
            return fvdb.ConvPackBackend.IGEMM

        elif backend == "igemm_mode1":
            kmap.build_implicit_gemm(
                sorted=True, split_mask_num=1, training=self.training, split_mask_num_bwd=3, use_tf32=self.allow_tf32
            )
            return fvdb.ConvPackBackend.IGEMM

        elif backend == "igemm_mode2":
            kmap.build_implicit_gemm(
                sorted=True, split_mask_num=3, training=self.training, split_mask_num_bwd=3, use_tf32=self.allow_tf32
            )
            return fvdb.ConvPackBackend.IGEMM
```

However, the build functions can only be called once, and if they're called again with a different setting, they'll raise a RuntimeException:

```

void
SparseConvPackInfo::buildImplicitGEMM(
    bool sorted, int splitMaskNum, bool training, int splitMaskNumBwd, bool use_tf32) {
    if (mIGEMMOutInMap.has_value()) {
        if (mIGEMMReorderLoc.has_value()) {
            TORCH_CHECK(mIGEMMReorderLoc->size(0) == splitMaskNum,
                        "Implicit GEMM is already built with different splitMaskNum value");
        }
        TORCH_CHECK(mIGEMMOutInMapBwd.has_value() == training,
                    "Implicit GEMM is already built with different training flag");
        return;
    }
```

I don't have a workaround yet, I think for now we have to reconstruct the pack info. I cannot tell from the code whether it is problematic to just always construct in 'training' mode.
